src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -1,19 +1,52 @@
 package core.basesyntax;
 
 /**
- * Реалізуйте методи для знаходження головної діагоналі прямокутної матриці,
- * максимального та мінімального елементів.
+ * Реалізуйте методи для знаходження головної діагоналі прямокутної матриці, максимального та
+ * мінімального елементів.
  */
 public class MatrixOperation {
+
+    /**
+     * Method return main diagonal of matrix.
+     * @return main diagonal of matrix
+     */
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] mainDiagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            mainDiagonal[i] = matrix[i][i];
+        }
+        return mainDiagonal;
     }
 
+    /**
+     * Method return min value of matrix.
+     * @return min value of matrix.
+     */
     public int getMin(int[][] matrix) {
-        return 0;
+        int min = Integer.MAX_VALUE;
```

Comment:
why we use Integer.MAX_VALUE here?

Reply:
Why not? We can use first element of this matrix, but use maximum value for this type its correct

Reply:
fixed